src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1702,24 +1702,23 @@ public final void stop() {
     public void interrupt() {
         if (this != Thread.currentThread()) {
             checkAccess();
-
+        }
+        // Write interrupted before reading nioBlocker for correct synchronization.
```

Comment:
I think I'd prefer if this said that it sets the interrupt status, and must be done before reading the nioBlocker.